sirepo/job_driver/__init__.py

```diff
@@ -101,13 +102,17 @@ def cpu_slot_free_one(self):
     async def cpu_slot_ready(self):
         if self.cpu_slot:
             return
+        if self._cpu_slot_alloc_lock.locked():
```

Comment:
All this can be encapsulated in the new abstraction. The only reason we have it literally is that we don't have good abstractions from tornado so we have to fix them up individually.

I think all Queues probably could use the same object that would manage the resource was allocated. 

I need to think about what the proper abstraction shoudl be.